zebrad/tests/acceptance.rs

```diff
@@ -461,3 +464,119 @@ fn valid_generated_config(command: &str, expected_output: &str) -> Result<()> {
 
     Ok(())
 }
+
+#[tokio::test]
+async fn metrics_endpoint() -> Result<()> {
+    use hyper::{Client, Uri};
+
+    zebra_test::init();
+
+    // [Note on port conflict](#Note on port conflict)
+    let endpoint = "127.0.0.1:50001";
+    let url = "http://127.0.0.1:50001";
+
+    // Write a configuration that has metrics endpoint_addr set
+    let mut config = default_test_config()?;
+    config.metrics.endpoint_addr = Some(endpoint.parse().unwrap());
+
+    let dir = TempDir::new("zebrad_tests")?;
+    fs::File::create(dir.path().join("zebrad.toml"))?
+        .write_all(toml::to_string(&config)?.as_bytes())?;
+
+    let tempdir = dir.path().to_path_buf();
+
+    let mut child = get_child(&["start"], &tempdir)?;
+
+    // Run the program for a second before testing the endpoint
+    std::thread::sleep(Duration::from_secs(1));
+
+    // Create an http client
+    let client = Client::new();
+
+    // Test metrics endpoint
+    let res = client.get(Uri::from_static(url)).await?;
+    assert!(res.status().is_success());
+    let body = hyper::body::to_bytes(res).await?;
+    assert!(std::str::from_utf8(&body)
+        .unwrap()
+        .contains("metrics snapshot"));
+
+    child.kill()?;
+
+    let output = child.wait_with_output()?;
+    let output = output.assert_failure()?;
+
+    // Make sure metrics was started
+    output.stdout_contains(format!(r"Initializing metrics endpoint at {}", endpoint).as_str())?;
+
+    // Make sure the command was killed
+    output.assert_was_killed()?;
```

Comment:
Let's add a useful hint to the error message.

We'll need to use eyre::WrapErr to use this function.

```suggestion
    output.assert_was_killed().wrap_err("Possible port conflict. Are there other acceptance tests running?")?;
```

Reply:
https://github.com/ZcashFoundation/zebra/pull/1000/commits/4e6bca466a78290c4a4d9e2906521cdcaf66ca54